apps/frontend/src/store/data_filters.ts

```diff
@@ -353,18 +374,186 @@ export const FilteredDataModule = getModule(FilteredData);
  * - defaulting "omit_overlayed_controls"
  */
 export function filter_cache_key(f: Filter) {
-  // fix the search term
-  let newSearch: string;
-  if (f.searchTerm !== undefined) {
-    newSearch = f.searchTerm.trim();
-  } else {
-    newSearch = '';
-  }
-
   const newFilter: Filter = {
-    searchTerm: newSearch,
+    searchTerm: f.searchTerm?.trim() || '',
     omit_overlayed_controls: f.omit_overlayed_controls || false,
     ...f
   };
   return JSON.stringify(newFilter);
 }
+
+export function lowercaseAll(input: string | string[]): string | string[] {
+  if (typeof input === 'string') {
+    return input.toLowerCase();
+  } else {
+    return input.map((string) => {
+      return string.toLowerCase();
+    });
+  }
+}
```

Comment:
This function doesn't seem to be used anywhere

Reply:
This is used in search.ts, moving to there instead.